deepchecks/tabular/checks/model_evaluation/weak_segments_performance.py

```diff
@@ -0,0 +1,201 @@
+            columns=[f'segment_score_{scorer.name}', 'feature1', 'feature1_range', 'feature2', 'feature2_range'])
+        for i in range(min(len(feature_rank_for_search), self.n_top_features)):
+            for j in range(i + 1, min(len(feature_rank_for_search), self.n_top_features)):
+                feature1, feature2 = feature_rank_for_search[[i, j]]
+                weak_segment_score, weak_segment_filter = self._find_weak_segment(dummy_model, encoded_dataset,
+                                                                                  [feature1, feature2], scorer,
+                                                                                  loss_per_sample)
+                weak_segments.loc[len(weak_segments)] = [round(weak_segment_score, 3), feature1,
+                                                         np.around(weak_segment_filter.filters[feature1], 3), feature2,
+                                                         np.around(weak_segment_filter.filters[feature2], 3)]
+        return weak_segments.sort_values(f'segment_score_{scorer.name}')
+
+    def _find_weak_segment(self, dummy_model, dataset, features_for_segment, scorer, loss_per_sample):
+        """Find weak segment based on scorer for specified features."""
+        if version.parse(sklearn.__version__) < version.parse('1.0.0'):
+            criterion = ['mse', 'mae']
+        else:
+            criterion = ['squared_error', 'absolute_error']
+        search_space = {
+            'max_depth': [5],
+            'min_weight_fraction_leaf': [self.segment_minimum_size_ratio],
+            'min_samples_leaf': [10],
+            'criterion': criterion
+        }
+
+        def get_worst_leaf_filter(tree):
+            leaves_filters = convert_tree_leaves_into_filters(tree, features_for_segment)
+            min_score, min_score_leaf_filter = np.inf, None
+            for leaf_filter in leaves_filters:
+                leaf_data = leaf_filter.filter(dataset.data)
+                leaf_score = scorer.run_on_data_and_label(dummy_model, leaf_data, leaf_data[dataset.label_name])
+                if leaf_score < min_score:
+                    min_score, min_score_leaf_filter = leaf_score, leaf_filter
+            return min_score, min_score_leaf_filter
+
+        def neg_worst_segment_score(clf: DecisionTreeRegressor, x, y) -> float:  # pylint: disable=unused-argument
+            return -get_worst_leaf_filter(clf.tree_)[0]
+
+        random_searcher = GridSearchCV(DecisionTreeRegressor(), scoring=neg_worst_segment_score,
+                                       param_grid=search_space, n_jobs=-1, cv=3)
+        try:
+            random_searcher.fit(dataset.features_columns[features_for_segment], loss_per_sample)
+            segment_score, segment_filter = get_worst_leaf_filter(random_searcher.best_estimator_.tree_)
+        except Exception as e:
+            raise DeepchecksProcessError('Unable to train meaningful error segmentation model') from e
```

Comment:
This exception is problematic - the user will have no idea what to do with it.
Also, this exception crashes the whole check, no? So if for 2 specific features you can't find a segmentation model, why crash the whole check instead of moving to the next couple?

Reply:
This is a very good point :)